### PR TITLE
Move the login and logout methods to a client.

### PR DESCRIPTION
**Pull request by elopio**

### CODE REVIEW

**Comment by elopio** [comment]
This is the first split of the macaroons branch. It just moves the simple login and logout methods to a client object. Requires #511.

**Comment by elopio** [comment]
retest this please

**Comment by sergiusens** [comment]
@elopio just a reminder to update the branch :wink:

**Comment by elopio** [comment]
The unit tests are passing locally and on jenkins, but failing on travis. And travis refuses to show the output. I'm retrying...

**Comment by sergiusens** [comment]
Oh, this is a case of cyclic dependencies. I had the same problem before. Travis uses python3.4 I believe and this is not handled vey well.

File: snapcraft/tests/store/test_store_client.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+
+class LoginTestCase(tests.TestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.useFixture(fixture_setup.FakeStore())
+        self.client = storeapi.StoreClient()
+
+    def test_login_successful(self):
```
Comment (sergiusens) [inline snapcraft/tests/store/test_store_client.py]: is there a fake server response where we use the correct password but otp is required? (just want to test the response)

**Comment by sergiusens** [comment]
:+1:

**Comment by elopio** [comment]
restest this please

File: snapcraft/tests/store/test_store_client.py (hunk 1 above)
Comment (elopio) [inline snapcraft/tests/store/test_store_client.py]: yes, look down:
test_failed_login_with_wrong_one_time_password

That's only in unit tests though. We could add one for integration, even hitting the staging test. I'm not sure if it adds value, but it's easy to do. File a bug if you want it, and assign it to me.

Comment (sergiusens) [inline snapcraft/tests/store/test_store_client.py]: I thought an incorrect otp was different than a missing one which is required

Comment (elopio) [inline snapcraft/tests/store/test_store_client.py]: as far as I know, the answer is the same. Either an sso.ApiException or an sso.UnexpectedApiException. That is handled by the ssoclient. From our side, we just need to make sure that the success field is False.
Unless you mean something different and I misunderstood your comment.

**Pull request merged**